File: tutorials/point_source_spiral/usr1.f
!vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvC
!                                                                      C
!  Module name: USR1                                                   C
!  Purpose: This routine is called from the time loop and is           C
!           user-definable.  The user may insert code in this routine  C
!           or call appropriate user defined subroutines.  This        C
!           can be used for setting or checking errors in quantities   C
!           that vary with time.  This routine is not called from an   C
!           IJK loop, hence all indices are undefined.                 C               C
!                                                                      C
!  Author:                                            Date: dd-mmm-yy  C
!  Reviewer:                                          Date: dd-mmm-yy  C
!                                                                      C
!  Revision Number:                                                    C
!  Purpose:                                                            C
!  Author:                                            Date: dd-mmm-yy  C
!  Reviewer:                                          Date: dd-mmm-yy  C
!                                                                      C
!  Literature/Document References:                                     C
!                                                                      C
!  Variables referenced:                                               C
!  Variables modified:                                                 C
!                                                                      C
!  Local variables:                                                    C
!                                                                      C
!^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^C
      SUBROUTINE USR1 

      use ps
      use constant
      use geometry
      use physprop
      use run
      use usr

      IMPLICIT NONE

      DOUBLE PRECISION :: lRad,   lTime
      INTEGER :: M

      lTime = time
      do while (lTime > 3.0d0)
         lTime = lTime - 3.0d0
      enddo

      lRad = 2.0d0*Pi*(lTime/3.0d0)

! Calculate the normalized velocity vectors for all phases.
      PS_V_g(1) = 0.12d0/0.15d0
      PS_U_g(1) = cos(lRad)/0.15d0
      PS_W_g(1) = sin(lRad)/0.15d0

      do M=1,MMAX
         PS_V_s(1,M) = 0.12d0/0.15d0
         PS_U_s(1,M) = 0.09d0*cos(lRad)/0.15d0
         PS_W_s(1,M) = 0.09d0*sin(lRad)/0.15d0
      enddo

      RETURN  
      END SUBROUTINE USR1

File: tutorials/point_source_spiral/usr1.py
# !vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvC
# !                                                                      C
# !  Module name: USR1                                                   C
# !  Purpose: This routine is called from the time loop and is           C
# !           user-definable.  The user may insert code in this routine  C
# !           or call appropriate user defined subroutines.  This        C
# !           can be used for setting or checking errors in quantities   C
# !           that vary with time.  This routine is not called from an   C
# !           IJK loop, hence all indices are undefined.                 C               C
# !                                                                      C
# !  Author:                                            Date: dd-mmm-yy  C
# !  Reviewer:                                          Date: dd-mmm-yy  C
# !                                                                      C
# !  Revision Number:                                                    C
# !  Purpose:                                                            C
# !  Author:                                            Date: dd-mmm-yy  C
# !  Reviewer:                                          Date: dd-mmm-yy  C
# !                                                                      C
# !  Literature/Document References:                                     C
# !                                                                      C
# !  Variables referenced:                                               C
# !  Variables modified:                                                 C
# !                                                                      C
# !  Local variables:                                                    C
# !                                                                      C
# !^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^C

from math import cos, sin, pi
from mfix import physprop as PHYSPROP
from mfix import ps as PS
from mfix import run as RUN

def usr1():

    # ! One revolution over a three second period.
    lRad = (2.0e0*pi*RUN.time)/3.0e0

    # ! Calculate the normalized velocity components.
    lV = (0.12e0/0.15e0)
    lU = (0.09e0/0.15e0)*cos(lRad)
    lW = (0.09e0/0.15e0)*sin(lRad)

    # ! Update the gas phase components.
    PS.ps_v_g[1-1] = lV
    PS.ps_u_g[1-1] = lU
    PS.ps_w_g[1-1] = lW

    # ! Update the solids phase components.
    for M in range(1, PHYSPROP.mmax+1):
        PS.ps_v_s[1-1, M-1] = lV
        PS.ps_u_s[1-1, M-1] = lU
        PS.ps_w_s[1-1, M-1] = lW
